LBHFSSPublicAPI/V1/Gateways/Interfaces/IServicesGateway.cs

```diff
@@ -8,6 +9,6 @@ namespace LBHFSSPublicAPI.V1.Gateways.Interfaces
     public interface IServicesGateway
     {
         ServiceEntity GetService(int id);
-        ICollection<ServiceEntity> SearchServices(SearchServicesRequest requestParams);
```

Comment:
Any particular reason why this was changed from a collection interface to a List?

Reply:
I don't quite remember, but I remember that it solved some issue in the factory. You can't do some things with `ICollection<T>` that you can with List.

But to give you you another good reason, it would be that the gateway was returning `List<ServiceEntity>` anyway, which made the `ICollection<T>` return type be pretentious and confusing. Here's a snippet of what we returned prior to my changes:

https://github.com/LBHackney-IT/lbh-fss-public-api/blob/develop/LBHFSSPublicAPI/V1/Gateways/ServicesGateway.cs#L68

Since we're doing `.ToList()` before returning the results, the return type will always be a List, so I figured: "Why pretend that we're going to return anything other than a List?".

I think that `ICollection<T>` is something that should be used in a generic method, not in the regular method. It should be used when you write your method to accept and process different kinds of collections like Arrays, HashSets and Dictionaries. But in our case we're returning a plain `List<T>` and it's good we do as it simplifies the logic later down the chain.